pkg/awsutils/gen_vpc_ip_limits.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// +build ignore
+
+// This program generates contributors.go. It can be invoked by running
+// go generate
+package main
+
+import (
+	"log"
+	"os"
+	"sort"
+	"text/template"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+const ipLimitFileName = "pkg/awsutils/vpc_ip_resource_limit.go"
+
+type ENILimit struct {
+	InstanceType string
+	ENILimit     int64
+	IPLimit      int64
+}
+
+// Helper function to call the EC2 DescribeInstanceTypes API and generate the IP limit file.
+func main() {
+	// Get session
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	_, err := sess.Config.Credentials.Get()
+	if err != nil {
+		log.Fatal(err)
+	}
+	svc := ec2.New(sess)
+	describeInstanceTypeInput := &ec2.DescribeInstanceTypesInput{}
+
+	eniLimitMap := make(map[string]ENILimit)
+	fetchMoreInstanceTypes := true
```

Comment:
femto-nit: you could lose this variable and just do:

```go
for {
///...
                if output.NextToken == nil {
			break
		}
        	describeInstanceTypeInput = &ec2.DescribeInstanceTypesInput{
			NextToken: output.NextToken,
		}
}
```

Reply:
Thanks, much cleaner. Had some debug prints in that loop earlier.